mondrian/src/main/java/mondrian/olap/fun/CrossJoinFunDef.java

```diff
@@ -999,29 +999,27 @@ private static boolean checkData(
         Evaluator evaluator)
     {
         if (cnt < 0) {
-            // no measures found, use standard algorithm
-            if (measureSet.isEmpty()) {
-                Object value = evaluator.evaluateCurrent();
-                if (value != null
-                    && !(value instanceof Throwable))
-                {
-                    return true;
-                }
-            } else {
+            boolean found = false;
+            Object value = evaluator.evaluateCurrent();
+            if (value != null
+                && !(value instanceof Throwable))
+            {
+                found = true;
+            }
+            if (!measureSet.isEmpty()) {
```

Comment:
Hey @Advent51 - I don't understand your fix.  Can you explain it some?  And actually, putting a little detail about how the change addresses the problem is more useful than a commit message that just says "Fixed."  :)